lib/singleton.rb

```diff
@@ -112,40 +112,31 @@ def _dump(depth = -1)
   module SingletonClassMethods # :nodoc:
 
     def clone # :nodoc:
-      Singleton.__init__(super)
+      super.include(Singleton)
     end
 
     # By default calls instance(). Override to retain singleton state.
-    def _load(str)
+    def _load(*)
```

Comment:
This is not a good change, as it changes the arity, and results in an unnecessary array allocation per call.